### PR TITLE
Ryan dev7 Restructure Redux states

### PR DESCRIPTION
**Pull request by ryanwooa**
- Restructure Redux states
- show completed tasks

### CODE REVIEW

File: components/application/widgets/CheckBoxButton.js [hunk 1]
```diff
@@ -16,7 +16,14 @@ import Icon from "./Icon";
...
 
-const CheckBoxButton = ({ priority, taskId, dueDate, projectId }) => {
+const CheckBoxButton = ({
+  priority,
+  taskId,
+  dueDate,
+  projectId,
+  completed,
+}) => {
+  const [isCompleted, setIsCompleted] = React.useState(completed);
```
Comment (zli213) [inline components/application/widgets/CheckBoxButton.js:26]: Why not import `react` and only use useState ?

File: components/application/widgets/CheckBoxButton.js [hunk 2]
```diff
@@ -42,22 +50,33 @@ const CheckBoxButton = ({ priority, taskId, dueDate, projectId }) => {
       });
 
       const result = await res.json();
-      console.log(result);
-      result.ifComplete &&
-        dispatch(completeTaskAction({ taskId, dueDate, projectId }));
+      result.ifComplete && dispatch(completeTaskAction(taskId));
+      setIsCompleted(result.ifComplete);
     } catch (error) {
       throw error;
```
Comment (zli213) [inline components/application/widgets/CheckBoxButton.js:56]: I recommend using "throw new Error('XXX ' + error.message)" to facilitate future error localization.

File: components/application/widgets/MyProjectItem.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ import {
 } from "../../../public/CommonFunctions";
 import { useRouter } from "next/navigation";
 import { useDispatch } from "react-redux";
-import { deleteProjectAction, archiveProjectAction, unarchiveProjectAction} from "../../../store/tasks";
+import { deleteProjectAction, archiveProjectAction,unarchiveProjectAction } from "../../../store/projects";
```
Comment (zli213) [inline components/application/widgets/MyProjectItem.js:15]: Maintain the spacing before 'unarchiveProjectAction' consistent with the previous version, that is, with a space before it .

File: components/application/widgets/SingleItem.js [hunk 4]
```diff
@@ -97,34 +98,29 @@ export function SingleItems({
...
-        updateTaskAction({
-          task: result.body,
-          oldDue: dueDate,
-          oldProjectId: projectId,
-        })
-      );
+      dispatch(updateTaskAction(result.body, dueDate, projectId));
 
-      //console.log(result.body);
+      console.log(result.body);
```
Comment (zli213) [inline components/application/widgets/SingleItem.js:103]: Need it?

File: components/application/widgets/TodoList.js [hunk 5]
```diff
@@ -147,25 +155,22 @@ function TodoList({
         }),
       });
       const result = await res.json();
-      console.log(result)
+      console.log(result);
```
Comment (zli213) [inline components/application/widgets/TodoList.js:158]: See above.

File: components/application/widgets/TodoList.js [hunk 6]
```diff
@@ -147,25 +155,22 @@ function TodoList({
...
-        dispatch(
-          editBoardAction({ board, projectId: fromProject.projectId, oldBoard: title })
-        );
+        dispatch(editBoardAction(fromProject.projectId, board, title));
         switchEditSectionHandler();
       } else if (result.body === "exist") {
         failEditHandler();
       }
     } catch (error) {
       throw error;
```
Comment (zli213) [inline components/application/widgets/TodoList.js:167]: See above.

File: styles/scss/application.module.scss [hunk 7]
```diff
@@ -364,3 +358,83 @@
...
+  left: 0;
+  right: 0;
+  bottom: 0;
+  border-radius: 15px;
+  background-color: $grayColor;
+  transition: 350ms ease all;
+}
+
+.view_checkbox {
+  border-radius: 50%; /* 设置开关内部图标为圆形 */
```
Comment (zli213) [inline styles/scss/application.module.scss:416]: Chinese

**Review by zli213 — APPROVED** [review]
Kelvin probably won't have time to review the code this time.

**Review by kesesek — APPROVED** [review]

File: components/application/widgets/CheckBoxButton.js (hunk 1 above)
Comment (ryanwooa) [reply components/application/widgets/CheckBoxButton.js]: I didnt notice this. May this part is written by Copilot. Fixed

File: components/application/widgets/CheckBoxButton.js (hunk 2 above)
Comment (ryanwooa) [reply components/application/widgets/CheckBoxButton.js:60]: Fixed

File: components/application/widgets/MyProjectItem.js (hunk 3 above)
Comment (ryanwooa) [reply components/application/widgets/MyProjectItem.js]: OK

File: components/application/widgets/SingleItem.js (hunk 4 above)
Comment (ryanwooa) [reply components/application/widgets/SingleItem.js]: No

File: components/application/widgets/TodoList.js (hunk 5 above)
Comment (ryanwooa) [reply components/application/widgets/TodoList.js]: Ok

File: components/application/widgets/TodoList.js (hunk 6 above)
Comment (ryanwooa) [reply components/application/widgets/TodoList.js:166]: Change the parameters

File: styles/scss/application.module.scss (hunk 7 above)
Comment (ryanwooa) [reply styles/scss/application.module.scss]: Deleted

**Pull request merged**